lib/valid_email2/dns.rb

```diff
@@ -67,5 +87,14 @@ def resolv_config
       config[:nameserver] = @dns_nameserver if @dns_nameserver
       config
     end
+
+    def generate_mx_cache_key(domain, domain_list, mx_servers)
+      return nil if mx_servers.empty? || domain_list.empty?
```

Comment:
In ruby, `nil` is the default return value.

```suggestion
      return if mx_servers.empty? || domain_list.empty?
```